provider/lxd/environ_broker_test.go

```diff
@@ -350,7 +361,28 @@ func (s *environBrokerSuite) TestStartInstanceNoTools(c *gc.C) {
 	c.Assert(err, gc.ErrorMatches, "no matching agent binaries available")
 }
 
+func (s *environBrokerSuite) TestStartInstanceInvalidCredentials(c *gc.C) {
+	c.Assert(s.invalidCredential, jc.IsFalse)
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+	svr := lxd.NewMockServer(ctrl)
+
+	exp := svr.EXPECT()
+	gomock.InOrder(
+		exp.HostArch().Return(arch.AMD64),
+		exp.FindImage("bionic", arch.AMD64, gomock.Any(), true, gomock.Any()).Return(containerlxd.SourcedImage{}, nil),
+		exp.GetNICsFromProfile("default").Return(s.defaultProfile.Devices, nil),
+		exp.CreateContainerFromSpec(gomock.Any()).Return(&containerlxd.Container{}, fmt.Errorf("not authorized")),
+	)
+
+	env := s.NewEnviron(c, svr, nil)
+	_, err := env.StartInstance(s.callCtx, s.GetStartInstanceArgs(c, "bionic"))
+	c.Assert(err, gc.ErrorMatches, "not authorized")
+	c.Assert(s.invalidCredential, jc.IsTrue)
+}
+
 func (s *environBrokerSuite) TestStopInstances(c *gc.C) {
+	c.Assert(s.invalidCredential, jc.IsFalse)
```

Comment:
Why is this here?
If we are checking that the credential is valid in the beginning of the test, I would have expected that the test makes some calls to invalidate the credential and then check that it is indeed invalid at the end. But it does not seem to be the case here :)

I think you wanted to make this call int he beginning of TestStopInstancesInvalidCredentials... It is missing from there :)